src/network/network_chat_gui.cpp

```diff
@@ -44,12 +45,18 @@ struct ChatMessage {
 };
 
 /* used for chat window */
-static ChatMessage *_chatmsg_list = nullptr; ///< The actual chat message list.
+static std::deque<ChatMessage> _chatmsg_list; ///< The actual chat message list.
 static bool _chatmessage_dirty = false;   ///< Does the chat message need repainting?
 static bool _chatmessage_visible = false; ///< Is a chat message visible.
 static bool _chat_tab_completion_active;  ///< Whether tab completion is active.
 static uint MAX_CHAT_MESSAGES = 0;        ///< The limit of chat messages to show.
 
+/**
+ * Time the chat history was marked dirty. This is used to determine if expired
+ * messages have recently expired and should cause a redraw to hide them.
+ */
+static std::chrono::steady_clock::time_point _chatmessage_dirty_time;
```

Comment:
Might be worth combining these all into a struct? Just to hold everything together